### PR TITLE
[Port] TL support for filtering

### PR DESCRIPTION
**Pull request by AbhitejJohn**
Porting this change from 15.4. Not porting specific commits to change th vsix for now since similar changes are already in master.

### CODE REVIEW

**Comment by genlu** [comment]
I ran into a stack overflow using the filter, filed an issue here: #1079

File: scripts/build/TestPlatform.Settings.targets [hunk 1]
```diff
@@ -54,6 +54,9 @@
       <PropertyGroup>
         <GenerateDocumentationFile>false</GenerateDocumentationFile>
       </PropertyGroup>
+      <ItemGroup>
```
Comment (smadala) [inline scripts/build/TestPlatform.Settings.targets]: Duplicate.

Comment (AbhitejJohn) [reply scripts/build/TestPlatform.Settings.targets]: looks like a bad merge. fixed!

**Comment by AbhitejJohn** [comment]
@smadala , @singhsarab .. you guys already approved this on another branch. If you don't see any concerns with this port, I can just push this in.

**Comment by genlu** [comment]
@AbhitejJohn When can we get this change merged and inserted?

**Comment by AbhitejJohn** [comment]
@dotnet-bot test Windows_NT / Release Build please.

**Comment by AbhitejJohn** [comment]
@genlu : Was hoping to get this in yesterday. Will push this in today after I get approvals.

**Review by smadala — APPROVED** [review]

**Review by singhsarab — APPROVED** [review]

**Pull request merged**